includes/managers/icons.php

```diff
@@ -377,25 +385,67 @@ public static function render_icon( $icon, $attributes = [], $tag = 'i' ) {
 	 *
 	 * @return array
 	 */
-	public static function fa4_to_fa5_value_migration( $value ) {
-		static $migration_dictionary = false;
-		if ( '' === $value ) {
-			return [
-				'value' => '',
+	public static function fa_icon_value_migration( $value ) {
+		static $migration_dictionary = [];
+
+		$mapping_files = self::get_mapping_files();
+		$found_in_mapping = false;
+
+		if ( is_string( $value ) ) {
+			if ( '' === $value ) {
+				return [
+					'value' => '',
+					'library' => '',
+				];
+			}
+
+			$value = [
+				'value' => $value,
 				'library' => '',
 			];
 		}
-		if ( false === $migration_dictionary ) {
-			$migration_dictionary = json_decode( Utils::file_get_contents( ELEMENTOR_ASSETS_PATH . 'lib/font-awesome/migration/mapping.js' ), true );
+
+		foreach ( $mapping_files as $mapping_file ) {
+			if ( ! isset( $migration_dictionary[ $mapping_file ] ) ) {
+				$migration_dictionary[ $mapping_file ] = json_decode( Utils::file_get_contents( $mapping_file ), true );
+			}
+
+			if ( isset( $migration_dictionary[ $mapping_file ][ $value['value'] ] ) ) {
+				$value = $migration_dictionary[ $mapping_file ][ $value ]['value'];
+				$found_in_mapping = true;
+			}
 		}
-		if ( isset( $migration_dictionary[ $value ] ) ) {
-			return $migration_dictionary[ $value ];
+
+		// Make sure there's no value in the old 'fa ' format, even if no replacement found.
+		if ( ! $found_in_mapping && '' === $value['library'] ) {
```

Comment:
Why do we need the " ! $found_in_mapping" flag?
Isn't it enough just to check if the library value is empty at that point?

Reply:
No, because of a very rare but possible scenario:
* the value is from an old version and in the old format.
* but there's no substitute found in any of the migration files.
 
 In this case we want to at least change it to the new format of:
'fa fa-something' => [ 'value' => 'fas fa-something', 'library' => 'fa-solid' ]

Reply:
You are right, removed.